src/wallet/fidelity.rs

```diff
@@ -146,17 +146,28 @@ impl FidelityBond {
     }
 
     /// Generate the bond's certificate hash.
-    pub(crate) fn generate_cert_hash(&self, addr: &str) -> sha256d::Hash {
+    pub(crate) fn generate_cert_hash(&self, addr: &str) -> Result<sha256d::Hash, FidelityError> {
         let cert_msg_str = format!(
             "fidelity-bond-cert|{}|{}|{}|{}|{}|{}",
-            self.outpoint, self.pubkey, self.cert_expiry, self.lock_time, self.amount, addr
+            self.outpoint,
+            self.pubkey,
+            self.cert_expiry.ok_or(FidelityError::BondDoesNotExist)?, // TODO: Should We panic or propagate the error here?
+            self.lock_time,
+            self.amount,
+            addr
```

Comment:
Instead of error here a more intuitive pattern would be returning Option. As it checks another option value so the function can return None, if that option is None. We don't need to send unconfirmed fidelity bond ever, so we should be able to safely `expect()` in most call sites.